apps/beeai-ui/src/modules/settings/SettingsView.tsx

```diff
@@ -6,42 +6,54 @@
 'use client';
 
 import { Tab, TabList, TabPanel, TabPanels, Tabs } from '@carbon/react';
+import { ComponentType, useMemo } from 'react';
 
 import { Container } from '#components/layouts/Container.tsx';
 import { MainContent } from '#components/layouts/MainContent.tsx';
 import { ViewHeader } from '#components/ViewHeader/ViewHeader.tsx';
 import { ViewStack } from '#components/ViewStack/ViewStack.tsx';
+import { useAppConfig } from '#contexts/AppConfig/index.ts';
 import { ProvidersView } from '#modules/providers/components/ProvidersView.tsx';
 import { VariablesView } from '#modules/variables/components/VariablesView.tsx';
+import { FeatureName } from '#utils/feature-flags.ts';
 
 import { ThemeView } from './ThemeView';
 
 export function SettingsView() {
+  const { featureFlags } = useAppConfig();
+
+  const items = useMemo(
+    () => ITEMS.filter(({ featureName }) => !featureName || featureFlags[featureName]),
+    [featureFlags],
+  );
+
   return (
     <MainContent>
       <Container size="lg">
         <ViewStack>
           <ViewHeader heading="Settings" />
           <Tabs>
             <TabList>
-              <Tab>Variables</Tab>
-              <Tab>Agent providers</Tab>
-              <Tab>Theme</Tab>
+              {items.map(({ title }) => (
+                <Tab key={title}>{title}</Tab>
+              ))}
             </TabList>
             <TabPanels>
-              <TabPanel>
-                <VariablesView />
-              </TabPanel>
-              <TabPanel>
-                <ProvidersView />
-              </TabPanel>
-              <TabPanel>
-                <ThemeView />
-              </TabPanel>
+              {items.map(({ title, component: Component }) => (
+                <TabPanel key={title}>
+                  <Component />
+                </TabPanel>
+              ))}
             </TabPanels>
           </Tabs>
         </ViewStack>
       </Container>
     </MainContent>
   );
 }
+
+const ITEMS: { title: string; component: ComponentType; featureName?: FeatureName }[] = [
+  { title: 'Variables', component: VariablesView, featureName: 'Variables' },
+  { title: 'Agent providers', component: ProvidersView, featureName: 'Providers' },
+  { title: 'Theme', component: ThemeView },
+];
```

Comment:
Maybe we could do just this:

```ts
const featureFlagsToDisplay: Partial<FeatureFlags> = {
  Variables: true,
};

type Items = /* ... */

const ITEMS: Items = [
  ...(featureFlagsToDisplay.Variables
    ? [{ title: 'Variables', component: VariablesView }]
    : []),
  { title: 'Theme', component: ThemeView },
];
```

Basically we use same exact object structure `FeatureFlags` has, so we don't have to extract keys type, and can use it to control what to show or not to show in the UI. We can also drop the `useMemo`.